data/scripts/dependency-resolver.py

```diff
@@ -540,7 +540,7 @@ def log_environment(context):
 
     args = parser.parse_args()
 
-    if args.compiler != default_compiler and not which(args.compiler):
+    if args.compiler != default_compiler and not which(args.compiler.split(" ")[0]):
```

Comment:
[minor] You could use `split()`, if it's possible that tabs are present in the variable.